doc/melange.rst

```diff
@@ -0,0 +1,291 @@
+    
+  - By default, all stanzas will have their targets attached to an alias
+    ``melange``. The behavior of this default alias is exclusive: if an alias
+    is explicitly defined in the stanza, the targets from this stanza will
+    be excluded from the ``melange`` alias.
+
+- ``(module_systems <module_systems>)`` specifies the JavaScript import and
+  export format used. The values allowed for ``<module_systems>`` are ``es6``
+  and ``commonjs``.
+
+  - ``es6`` will follow `JavaScript modules <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules>`_,
+    and will produce ``import`` and ``export`` statements.
+
+  - ``commonjs`` will follow `CommonJS modules <https://nodejs.org/api/modules.html>`_,
+    and will produce `require` calls and export values with ``module.exports``.
+
+  - If no extension is specified, the resulting JavaScript files will use
+    ``.js``. You can specify a different extension with a pair
+    ``(<module_system> <extension>)``, e.g. ``(module_systems (es6 mjs))``.
+
+  - Multiple module systems can be used in the same field as long as their
+    extensions are different. For example,
+    ``(module_systems commonjs (es6 mjs))`` will produce one set of JavaScript
+    files using CommonJS and the ``.js`` extension, and another using ES6 and
+    the ``.mjs`` extension.
+
+- ``(modules <modules>)`` specifies what modules will be built with Melange. By
+  default, if this field is not defined, Dune will use all the ``.ml/.re`` files
+  in the same directory as the ``dune`` file. This includes module sources
+  present in the file system as well as modules generated by user rules. You can
+  restrict this list by using an explicit ``(modules <modules>)`` field.
+  ``<modules>`` uses the :doc:`reference/ordered-set-language`, where elements
+  are module names and don't need to start with an uppercase letter. For
+  instance, to exclude module ``Foo``, use ``(modules :standard \ foo)``.
+
+- ``(libraries <library-dependencies>)`` specifies Melange library dependencies.
+  Melange libraries can only use the simple form, like
+  ``(libraries foo pkg.bar)``. Keep in mind the following limitations:
+
+  - The ``re_export`` form is not supported.
+
+  - All the libraries included in ``<library-dependencies>`` have to support
+    the ``melange`` mode (see the section about libraries below).
+
+
+- ``(package <package>)`` allows the user to define the JavaScript package to
+  which the artifacts produced by the ``melange.emit`` stanza will belong.
+
+- ``(runtime_deps <paths-to-deps>)`` specifies dependencies that should be
+  copied to the build folder together with the ``.js`` files generated from the
+  sources. These runtime dependencies can include assets like CSS files, images,
+  fonts, external JavaScript files, etc. ``runtime_deps`` adhere to the formats
+  in :doc:`concepts/dependency-spec`. For example
+  ``(runtime_deps ./path/to/file.css (glob_file_rec ./fonts/*))``.
```

Comment:
this should be `glob_files_rec`

Reply:
Thanks! Fixed in https://github.com/ocaml/dune/pull/7477/commits/5313d5f743f534f04f4d7481f1704a1a84b6bd6f.